src/Razor/src/Microsoft.AspNetCore.Razor.Language/RazorCodeDocumentExtensions.cs

```diff
@@ -194,6 +196,50 @@ public static void SetFileKind(this RazorCodeDocument document, string fileKind)
             document.Items[typeof(FileKinds)] = fileKind;
         }
 
+        internal static bool TryComputeBaseNamespace(this RazorCodeDocument document, out string baseNamespace)
```

Comment:
Any reason to have this as an extension method on code document vs a private method in this class?

Reply:
I think an extension method make sense - we've already established that code document is a generic *container*.

Reply:
It's not used elsewhere currently but it seems like a nice to have extension method. I don't feel strongly either way.